grr/server/grr_response_server/flows/general/osquery_test.py

```diff
@@ -261,11 +268,182 @@ def testTotalRowCountIncludesAllChunks(self):
 
     with test_lib.ConfigOverrider({"Osquery.max_chunk_size": chunk_bytes}):
       with osquery_test_lib.FakeOsqueryiOutput(stdout=table_json, stderr=""):
-        flow_id = self._InitializeFlow("doesn't matter")
+        flow_id = self._InitializeFlow("doesn't matter", [])
         progress = flow_test_lib.GetFlowProgress(self.client_id, flow_id)
 
     self.assertEqual(progress.total_row_count, row_count)
 
+  def testFlowTruncatesFileAboveSingleLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      target_bytes = 2**20  # 1 MiB
+      less_than_necessary_bytes = target_bytes // 2
+
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write(b"1" * target_bytes)
+
+      table = f"""
+      [
+        {{ "collect_collumn": "{temp_file_path}" }}
+      ]
+      """
+
+      with mock.patch.object(
+          osquery_flow,
+          "FILE_COLLECTION_MAX_SINGLE_FILE_BYTES",
+          less_than_necessary_bytes):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          with self.assertRaises(RuntimeError):
+            self._RunFlow("Doesn't matter", ["collect_collumn"])
+
+  def testFlowDoesntCollectSingleFileAboveTotalLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      target_bytes = 2**20  # 1 MiB
+      less_than_necessary_bytes = target_bytes // 2
+
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write(b"1" * target_bytes)
+
+      table = f"""
+      [
+        {{ "collect_collumn": "{temp_file_path}" }}
+      ]
+      """
+
+      with mock.patch.object(
+          osquery_flow,
+          "FILE_COLLECTION_MAX_TOTAL_BYTES",
+          less_than_necessary_bytes):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          with self.assertRaises(RuntimeError):
+            self._RunFlow("Doesn't matter", ["collect_collumn"])
+
+  def testFlowCollectFile(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write("Just sample text to put in the file.".encode('utf-8'))
```

Comment:
You can also open the file using the UTF-8 encoding and write string directly. I find it a bit more natural:

```python
with io.open(temp_file_path, mode="w", encoding="utf-8") as fd:
    fd.write("foobar")
```

Reply:
Ah, terrible GitHub review flow strikes again.

No strong opinion here, can stay as it is.